src/test/java/SixPMComTestSuite/SixPMFeaturesTest.java

```diff
@@ -0,0 +1,88 @@
+package SixPMComTestSuite;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import java.util.List;
+import java.util.Random;
+
+import static SixPMComTestSuite.utils.SixPmConstants.*;
+
+public class SixPMFeaturesTest {
+    private WebDriver driver;
+
+    @BeforeClass
+    public void setDriver() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/chromedriver.exe");
+    }
+
+    @BeforeMethod
+    public void setChromeDriver(){
+        driver = new ChromeDriver();
+    }
+
+    @AfterMethod
+    public  void closeDriver(){
+        driver.quit();
+    }
+
+    @Test
+    public void addToFavTest(){
+        WebDriverWait wait = new WebDriverWait(driver,10);
+        Actions action = new Actions(driver);
+        Random random = new Random();
+
+        driver.manage().window().maximize();
+        driver.get(homePage);
+
+        WebElement accessoriesElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(accessoriesXPath)));
+        action.moveToElement(accessoriesElem).build().perform();
+
+        WebElement aviatorsButtonElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(aviatorsButtonXPath)));
+        aviatorsButtonElem.click();
+
+        List <WebElement> listOfAllItems = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(listOfAllItemsXPath)));
+
+        int countOfItems = listOfAllItems.size();
+        listOfAllItems.get(random.nextInt(countOfItems)).click();
+
+        WebElement addToBagElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(addToBagElemXPath)));
+        addToBagElem.click();
+
+        WebElement itemNameElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemNameElemActual)));
+        String actualItemName = itemNameElem.getText();
+        WebElement itemInfoElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemInfoElemActual)));
+        String actualItemInfo = itemInfoElem.getText();
+        WebElement itemPriceElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemPriceElemActual)));
+        String actualItemPrice = itemPriceElem.getText();
```

Comment:
The logic isn't correct. You're getting everything from bag and at the end you check equality of those with themselves.
When you click on 'View bag', it's just another view of shopping bag, but the items are the same. So you should at first get all your needed data, and then add product to shopping bag and at the end do assertions ( without clicking on 'View bag' button and redirecting to new page).

Reply:
changed